src/dptlib/index.ts

```diff
@@ -171,7 +171,7 @@ export function populateAPDU(value: any, apdu: APDU, dptid?: number | string) {
 	} else {
 		if (!isFinite(value)) {
 			throw new Error(
-				util.format('Invalid value, expected a %s', dpt.desc),
+				util.format('Invalid value, expected a %s', dpt.basetype?.desc),
```

Comment:
wondering if both `dpt.desc` and `dpt.basetype.desc` could be defined?

Reply:
Maybe it could make sense to put an or there and print one or the other based on the one that is defined

Reply:
Hi
It was an ancient mistake, because the dpt.desc property should never exists out of the basetype object. :-)

Reply:
@Supergiovane Ok so I suggest to fix the type and drop it on root :)